<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>GroSeq Analysis</title>
<meta name="author" content="(Edmund Miller)"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/reveal.css"/>

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/theme/white.css" id="theme"/>


<!-- If the query includes 'print-pdf', include the PDF print sheet -->
<script>
    if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'https://cdn.jsdelivr.net/npm/reveal.js/css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
</script>
</head>
<body>
<div class="reveal">
<div class="slides">
<section>
<section id="slide-orgfd1a0c5">
<h2 id="orgfd1a0c5">GRO-Seq Analysis</h2>
</section>
</section>
<section>
<section id="slide-org94089c5">
<h2 id="org94089c5">What I&rsquo;ve been working on</h2>
<div class="outline-text-2" id="text-org94089c5">
</div>
</section>
</section>
<section>
<section id="slide-orgbfd7447">
<h3 id="orgbfd7447">IMR90</h3>
<div class="outline-text-3" id="text-orgbfd7447">
</div>
</section>
<section id="slide-org7a85655">
<h4 id="org7a85655">Overview of Project for new Members</h4>
<ul>
<li>Compared IMR90 Predicted Enhancers to GM</li>
<li>Ran GREAT Analysis on them</li>
<li>Began working to find coverage of eRNAs in Samples</li>

</ul>
</section>
<section id="slide-org8cf6eaa">
<h4 id="org8cf6eaa">Compared IMR90 Predicted Enhancers to GM</h4>
<div class="outline-text-4" id="text-org8cf6eaa">
</div>
<ul class="org-ul">
<li><a id="orga3b0b2a"></a>Venn Diagram<br />
<div class="org-src-container">

<pre  class="src src-python"><code trim><span style="color: #51afef;">import</span> pylab <span style="color: #51afef;">as</span> plt
<span style="color: #51afef;">from</span> matplotlib_venn <span style="color: #51afef;">import</span> venn2
venn2(subsets = (<span style="color: #da8548; font-weight: bold;">6259</span>, <span style="color: #da8548; font-weight: bold;">10051</span>, <span style="color: #da8548; font-weight: bold;">1108</span>), set_labels = (<span style="color: #98be65;">'IMR90'</span>, <span style="color: #98be65;">'GM'</span>))
plt.title(<span style="color: #98be65;">"IMR90 vs GM eRNAs"</span>)
plt.show()
</code></pre>
</div>
</li>
</ul>
</section>
<section id="slide-org59cb2ba" data-background="img/Venn.png" data-background-size="800px" data-background-transition="slide">
<h4 id="org59cb2ba"></h4>
</section>
<section id="slide-org04afc17">
<h4 id="org04afc17">GREAT IMR90 Unique</h4>
<ul>
<li>lung vasculature development</li>
<li>growth factor receptor binding</li>
<li>tissue remodeling</li>
<li>positive regulation of endothelial cell migration</li>
<li>embryonic skeletal joint development</li>

</ul>
</section>
<section id="slide-orgff1d82d">
<h4 id="orgff1d82d">GREAT IMR90 and GM Overlap</h4>
<ul>
<li>643(~30%) Transcripts were -500 to -50 from TSS</li>
<li>1002(80%) Were Associated with 2 genes per region</li>
<li>104(10%) were associated with 1 gene</li>
<li>Notable Biological processes:
<ul>
<li>cytokine secretion</li>
<li>cytokine production</li>
<li>regulation of immune response</li>
<li>immune effector process</li>
<li>regulation of transcription from RNA polymerase II promoter in response to stress</li>

</ul></li>

</ul>
</section>
<section id="slide-org5db8e6c">
<h4 id="org5db8e6c">Coverage of eRNAs in Samples</h4>
<ul>
<li>Issue with lack of output</li>
<li>Cause is most likely file types</li>

</ul>
</section>
<section id="slide-org924e2cf">
<h4 id="org924e2cf">Snakemake IMR90</h4>
</section>
<section id="slide-org5902697" data-background="img/dag.png" data-background-size="400px" data-background-transition="slide">
<h4 id="org5902697"></h4>
</section>
<section id="slide-orgca41ecd">
<h4 id="orgca41ecd">Trying to Reproduce GM</h4>
<div class="outline-text-4" id="text-orgca41ecd">
</div>
<ul class="org-ul">
<li><a id="org714b3af"></a>hg18 vs hg19<br /></li>
<li><a id="org6baf093"></a>Overpredicting eRNA transcripts<br /></li>
</ul>
</section>
</section>
<section>
<section id="slide-orgbbe600a">
<h3 id="orgbbe600a">List of Super Enhancers</h3>
<div class="outline-text-3" id="text-orgbbe600a">
</div>
</section>
<section id="slide-orgb329fbc">
<h4 id="orgb329fbc">dbSuper</h4>
<p>
<a href="http://asntech.org/dbsuper/ses.php?genome=hg19&amp;cell_type=C_058">http://asntech.org/dbsuper/ses.php?genome=hg19&amp;cell_type=C_058</a>
</p>
</section>
</section>
<section>
<section id="slide-org6976c37">
<h3 id="org6976c37"><a href="https://github.com/r-bioinformatics/edgePy">edgePy</a></h3>
<div class="outline-text-3" id="text-org6976c37">
</div>
</section>
<section id="slide-org367b132">
<h4 id="org367b132">What is edgeR</h4>
<div class="outline-text-4" id="text-org367b132">
</div>
<ul class="org-ul">
<li><a id="orgcc0db1f"></a>Written in 2009<br /></li>
<li><a id="orge3a2c6e"></a>digital gene expression (DGE)<br /></li>
<li><a id="org9ee5c05"></a>Package<br />
<ul>
<li>For examining differential expression of replicated count data</li>

</ul>
</li>
</ul>
</section>
<section id="slide-orgf2e977b">
<h4 id="orgf2e977b">Contributions</h4>
<div class="outline-text-4" id="text-orgf2e977b">
</div>
<ul class="org-ul">
<li><a id="org6c5a4b7"></a>Updating Documentation<br /></li>
<li><a id="orge5b5e1e"></a>Added black formater<br /></li>
</ul>
</section>
</section>
<section>
<section id="slide-org6f9027a">
<h2 id="org6f9027a">Reproducibility</h2>
<div class="outline-text-2" id="text-org6f9027a">
</div>
</section>
</section>
<section>
<section id="slide-orgb3871f4">
<h3 id="orgb3871f4">Copy .bash _history</h3>
<div class="outline-text-3" id="text-orgb3871f4">
</div>
</section>
<section id="slide-org899903c">
<h4 id="org899903c">All the commands for GroSeq</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>fastqc *.fastq -o=/media/enhancer/IMR90/fastqc -t=<span style="color: #da8548; font-weight: bold;">6</span>
<span style="color: #51afef;">for</span> i<span style="color: #51afef;"> in</span> <span style="color: #51afef; font-weight: bold;">`</span><span style="color: #51afef; font-weight: bold;">ls</span><span style="color: #51afef; font-weight: bold;"> *.fastq | cut -d "_" -f 1`</span> ;
    <span style="color: #51afef;">do</span> bowtie2 -t -p <span style="color: #da8548; font-weight: bold;">4</span> -x ../2018-06-24/genome $<span style="color: #dcaeea;">i</span> -S $<span style="color: #dcaeea;">i</span>.sam ;
<span style="color: #51afef;">done</span> ;
<span style="color: #ECBE7B;">mv</span> data/2018-06-23/*.sam results/2018-07-10/
makeTagDirectory results/2018-07-11/All_together/ -genome hg19 -checkGC results/2018-07-10/*.sam
findPeaks results/2018-07-11/All_together/ -o results/2018-07-12/IMRpeak.gtf -style groseq
pos2bed.pl results/2018-07-12/GMpeak.gtf &gt; results/2018-07-12/GMpeak.bed
slopBed -i refseqhg19.bed -g Homo_sapiens/ucsc/hg19/Seqence/WholeGenomeFasta/genome.fa.fai -l <span style="color: #da8548; font-weight: bold;">1000</span> -r <span style="color: #da8548; font-weight: bold;">10000</span> &gt; slop-refseqhq19.bed
bedtools intersect -a GM/GMGRO-Seq/IMR90tsspeak.bed -b slop-refseqhg19.bed -v &gt; noGenesIMR90tss.bed
intersectBed -wo -a IMR90/BED/sorted_GroseqIMR90nogenes.bed -b /media/enhancer/eRNA_GM_hg19.bed &gt; eRNA_IMR90_GM_hg19_Overlap.bed
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org14c066d">
<h3 id="org14c066d">Jupyter</h3>
<p>
<a href="https://mybinder.org/v2/gh/widdowquinn/Notebooks-Bioinformatics/master">https://mybinder.org/v2/gh/widdowquinn/Notebooks-Bioinformatics/master</a>
</p>
</section>
</section>
<section>
<section id="slide-orgd5947c6">
<h3 id="orgd5947c6">Google Colab</h3>
<p>
<a href="https://colab.research.google.com/notebooks/mlcc/intro_to_pandas.ipynb">https://colab.research.google.com/notebooks/mlcc/intro_to_pandas.ipynb</a>
</p>
</section>
</section>
<section>
<section id="slide-org344c59c">
<h3 id="org344c59c">Send it to me!</h3>
<div class="outline-text-3" id="text-org344c59c">
</div>
</section>
<section id="slide-orge8fcfa5">
<h4 id="orge8fcfa5">GitLab</h4>
</section>
</section>
<section>
<section id="slide-org842e996">
<h2 id="org842e996">Mendeley</h2>
</section>
</section>
</div>
</div>
<script src="https://cdn.jsdelivr.net/npm/reveal.js/js/reveal.js"></script>

<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({
multiplex: {
    secret: '', // null if client
    id: '', // id, obtained from socket.io server
    url: '' // Location of socket.io server
},

// Optional libraries used to extend on reveal.js
dependencies: [
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }]
});
</script>
</body>
</html>
