#+TITLE: GroSeq Analysis
#+OPTIONS: reveal_title_slide:nil
#+OPTIONS: num:nil
#+OPTIONS: toc:nil
#+REVEAL_THEME: white
#+REVEAL_ROOT: https://cdn.jsdelivr.net/npm/reveal.js
#+REVEAL_THEME: white
#+REVEAL_HLEVEL: 2
* GRO-Seq Analysis
* What I've been working on
** IMR90
*** Overview of Project for new Members
- Compared IMR90 Predicted Enhancers to GM
- Ran GREAT Analysis on them
- Began working to find coverage of eRNAs in Samples
*** Compared IMR90 Predicted Enhancers to GM
**** Venn Diagram
#+BEGIN_SRC python
import pylab as plt
from matplotlib_venn import venn2
venn2(subsets = (6259, 10051, 1108), set_labels = ('IMR90', 'GM'))
plt.title("IMR90 vs GM eRNAs")
plt.show()
#+END_SRC
*** 
:PROPERTIES:
:reveal_background: img/Venn.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** GREAT IMR90 Unique
      - lung vasculature development
      - growth factor receptor binding
      - tissue remodeling
      - positive regulation of endothelial cell migration
      - embryonic skeletal joint development
*** GREAT IMR90 and GM Overlap
      - 643(~30%) Transcripts were -500 to -50 from TSS
      - 1002(80%) Were Associated with 2 genes per region
      - 104(10%) were associated with 1 gene
      - Notable Biological processes:
        - cytokine secretion
        - cytokine production
        - regulation of immune response
        - immune effector process
        - regulation of transcription from RNA polymerase II promoter in response to stress
*** Coverage of eRNAs in Samples
- Issue with lack of output
- Cause is most likely file types
*** Snakemake IMR90
***
:PROPERTIES:
:reveal_background: img/dag.png
:reveal_background_size: 400px
:reveal_background_trans: slide
:END:
*** Trying to Reproduce GM
**** hg18 vs hg19
**** Overpredicting eRNA transcripts
** List of Super Enhancers
*** dbSuper
http://asntech.org/dbsuper/ses.php?genome=hg19&cell_type=C_058
** [[https://github.com/r-bioinformatics/edgePy][edgePy]]
*** What is edgeR
**** Written in 2009
**** digital gene expression (DGE)
**** Package
- For examining differential expression of replicated count data
*** Contributions
**** Updating Documentation
**** Added black formater
* Reproducibility
** Copy .bash _history
*** All the commands for GroSeq
#+BEGIN_SRC shell
fastqc *.fastq -o=/media/enhancer/IMR90/fastqc -t=6
for i in `ls *.fastq | cut -d "_" -f 1` ;
    do bowtie2 -t -p 4 -x ../2018-06-24/genome $i -S $i.sam ;
done ;
mv data/2018-06-23/*.sam results/2018-07-10/
makeTagDirectory results/2018-07-11/All_together/ -genome hg19 -checkGC results/2018-07-10/*.sam
findPeaks results/2018-07-11/All_together/ -o results/2018-07-12/IMRpeak.gtf -style groseq
pos2bed.pl results/2018-07-12/GMpeak.gtf > results/2018-07-12/GMpeak.bed
slopBed -i refseqhg19.bed -g Homo_sapiens/ucsc/hg19/Seqence/WholeGenomeFasta/genome.fa.fai -l 1000 -r 10000 > slop-refseqhq19.bed
bedtools intersect -a GM/GMGRO-Seq/IMR90tsspeak.bed -b slop-refseqhg19.bed -v > noGenesIMR90tss.bed
intersectBed -wo -a IMR90/BED/sorted_GroseqIMR90nogenes.bed -b /media/enhancer/eRNA_GM_hg19.bed > eRNA_IMR90_GM_hg19_Overlap.bed
#+END_SRC
** Jupyter
https://mybinder.org/v2/gh/widdowquinn/Notebooks-Bioinformatics/master
** Google Colab
https://colab.research.google.com/notebooks/mlcc/intro_to_pandas.ipynb
** Send it to me!
*** GitLab
* Mendeley
