#+REVEAL_THEME: white
#+TITLE: Transcriptional Kinetics of Viral Induced eRNAs across cell lines
#+AUTHOR: Edmund Miller
#+OPTIONS: reveal_title_slide:nil
#+OPTIONS: num:nil
#+OPTIONS: toc:nil
#+REVEAL_ROOT: https://cdn.jsdelivr.net/npm/reveal.js
#+REVEAL_HLEVEL: 2
* Xonsh
** SciPy 2019
*** Links
**** [[http://conference.scipy.org/][Conference List]]
**** [[https://xonsh.github.io/scipy-2019-tutorial/#1][Xonsh Tutorial]]
**** [[https://youtu.be/ujo9sbqHFqw?list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC][Xonsh Talk]]
**** [[https://www.youtube.com/watch?v=ZB7BZMhfPgk&list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC][Talk Playlist]]
* Transcriptional Kinetics of Viral Induced enhancer-promoter pairs across cell lines
** Recap
*** 
:PROPERTIES:
:reveal_background: img/globaltrans.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** Overview
- Reproducing GM18
- Predicted IMR90 eRNAs
- Compared IMR90 Predicted Enhancers to GM
- Used Homer scripts to find DE of eRNAs and Genes
- Gene Centric vs. Enhancer Centric
*** Reproducing GM18
- hg18 vs hg19
- Overpredicting eRNA transcripts
- Past Issue
  - What I thought Peng sent me
  - hg18 -> eRNAs -> Me
  - What actually happened
  - hg18 -> eRNAs -> LiftOver -> hg19 -> Me
- Main issue is homer uniqmap
*** 
:PROPERTIES:
:reveal_background: img/dag.png
:reveal_background_size: 400px
:reveal_background_trans: slide
:END:
*** Enhancer Transcript Identification
Changes from GM18
- hg19
- No liftover
*** 
:PROPERTIES:
:reveal_background: img/venn/GM_eRNA_overlaps.svg
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/venn/GM19_eRNA_overlaps.svg
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/venn/IMR_eRNA_overlaps.svg
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/venn/eRNA_cross_cell.svg
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** Gene Centric vs. Enhancer Centric
- Peng's approach
  - Took enhancers that were expressed deferentially
  - Linked them to Genes within 200Kb
- New approach
  - Find genes that are deferentially expressed
  - Link the Enhancers to those genes

*** 
:PROPERTIES:
:reveal_background: viz/pipeline.png
:reveal_background_size: 600px
:reveal_background_trans: slide
:END:
*** featureCounts
***** Tried CLI version
***** Couldn't get examples to work with edgeR
***** Switched to R package
***** Switched to a snakemake wrapper
*** Limma
- [[https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/][RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR]]
*** 
:PROPERTIES:
:reveal_background: img/limma/GM19_fig3_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/IMR_fig3_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
** Overview
- Get List of inducible Genes
- Link inducible eRNAs and inducible genes
- Checking for L2
- Compare IMR and GM inducible lists
*** Get List of DE Genes
- Used TPM values
- Removed Genes without a foldchange > 1 between any timepoints
- Considering using only points t<24h for inducibility
- Working to implement one-step autocorrelation
*** Link eRNAs and Top genes
- Awk script to get RefSeq, with only specific BED entries
- Ended up using Ripgrep
#+BEGIN_SRC shell
awk -F \"\\t\" '{{ if (NR!=1){{ print $1 }}}}' genesFoldChange.txt | \
rg --dfa-size-limit 2G -w -f - refseqgenes.gtf | \
rg -w exon - > de_genes.gtf
#+END_SRC
- Linked eRNAs within 200Kb
*** Compare IMR and GM eRNAs
- Created Meta eRNAs
  - Find overlaps of IMR and GM (Viral)
  - Merge GM and IMR eRNAs(Meta)
  - Report Meta Viral eRNAs
*** 
:PROPERTIES:
:reveal_background: img/venn/eRNA_cross_cell.svg
:reveal_background_size: 1000px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/venn/eRNA_cross_cell_viral.svg
:reveal_background_size: 1000px
:reveal_background_trans: slide
:END:
*** Checking for L2
*** 
:PROPERTIES:
:reveal_background: img/L2_GenomeBrowser_v2.png
:reveal_background_size: 1000px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/peaks.groseqChart.png
:reveal_background_size: 1000px
:reveal_background_trans: slide
:END:
*** Possible Groupings
*** 
:PROPERTIES:
:reveal_background: img/grouping_eRNAs.png
:reveal_background_size: 1500px
:reveal_background_trans: slide
:END:
** [[https://emiller88.github.io/presentations/Functional_Genomics/2019-10-16/report.html][Snakemake Report]]
** Future
- Clean Groupings
- Tweak Homer
- Work with TNFSF10 to test grouping Identification Pipeline
