<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>Transcriptional Kinetics of Viral Induced eRNAs across cell lines</title>
<meta name="author" content="(Edmund Miller)"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/reveal.css"/>

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/theme/white.css" id="theme"/>


<!-- If the query includes 'print-pdf', include the PDF print sheet -->
<script>
    if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'https://cdn.jsdelivr.net/npm/reveal.js/css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
</script>
</head>
<body>
<div class="reveal">
<div class="slides">
<section>
<section id="slide-org064be22">
<h2 id="org064be22">Xonsh</h2>
<div class="outline-text-2" id="text-org064be22">
</div>
</section>
</section>
<section>
<section id="slide-org868b7c8">
<h3 id="org868b7c8">SciPy 2019</h3>
<div class="outline-text-3" id="text-org868b7c8">
</div>
</section>
<section id="slide-org92c2da1">
<h4 id="org92c2da1">Links</h4>
<div class="outline-text-4" id="text-org92c2da1">
</div>
<ul class="org-ul">
<li><a id="orgf94b2f2"></a><a href="http://conference.scipy.org/">Conference List</a><br /></li>
<li><a id="orge6c82e0"></a><a href="https://xonsh.github.io/scipy-2019-tutorial/#1">Xonsh Tutorial</a><br /></li>
<li><a id="orgc6b522e"></a><a href="https://youtu.be/ujo9sbqHFqw?list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC">Xonsh Talk</a><br /></li>
<li><a id="org43c2eff"></a><a href="https://www.youtube.com/watch?v=ZB7BZMhfPgk&amp;list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC">Talk Playlist</a><br /></li>
</ul>
</section>
</section>
<section>
<section id="slide-org1cec442">
<h2 id="org1cec442">Transcriptional Kinetics of Viral Induced enhancer-promoter pairs across cell lines</h2>
<div class="outline-text-2" id="text-org1cec442">
</div>
</section>
</section>
<section>
<section id="slide-org06490db">
<h3 id="org06490db">Recap</h3>
<div class="outline-text-3" id="text-org06490db">
</div>
</section>
<section id="slide-org374a8f5" data-background="img/globaltrans.png" data-background-size="800px" data-background-transition="slide">
<h4 id="org374a8f5"></h4>
</section>
<section id="slide-org66f78b2">
<h4 id="org66f78b2">Overview</h4>
<ul>
<li>Reproducing GM18</li>
<li>Predicted IMR90 eRNAs</li>
<li>Compared IMR90 Predicted Enhancers to GM</li>
<li>Used Homer scripts to find DE of eRNAs and Genes</li>
<li>Gene Centric vs. Enhancer Centric</li>

</ul>
</section>
<section id="slide-orgbcd7718">
<h4 id="orgbcd7718">Reproducing GM18</h4>
<ul>
<li>hg18 vs hg19</li>
<li>Overpredicting eRNA transcripts</li>
<li>Past Issue
<ul>
<li>What I thought Peng sent me</li>
<li>hg18 -&gt; eRNAs -&gt; Me</li>
<li>What actually happened</li>
<li>hg18 -&gt; eRNAs -&gt; LiftOver -&gt; hg19 -&gt; Me</li>

</ul></li>
<li>Main issue is homer uniqmap</li>

</ul>
</section>
<section id="slide-orgcd3bc8d" data-background="img/dag.png" data-background-size="400px" data-background-transition="slide">
<h4 id="orgcd3bc8d"></h4>
</section>
<section id="slide-org89b5cd4">
<h4 id="org89b5cd4">Enhancer Transcript Identification</h4>
<p>
Changes from GM18
</p>
<ul>
<li>hg19</li>
<li>No liftover</li>

</ul>
</section>
<section id="slide-orga28f04e" data-background="img/venn/GM_eRNA_overlaps.svg" data-background-size="800px" data-background-transition="slide">
<h4 id="orga28f04e"></h4>
</section>
<section id="slide-orgc3d298d" data-background="img/venn/GM19_eRNA_overlaps.svg" data-background-size="800px" data-background-transition="slide">
<h4 id="orgc3d298d"></h4>
</section>
<section id="slide-org17c538b" data-background="img/venn/IMR_eRNA_overlaps.svg" data-background-size="800px" data-background-transition="slide">
<h4 id="org17c538b"></h4>
</section>
<section id="slide-orgc511537" data-background="img/venn/eRNA_cross_cell.svg" data-background-size="800px" data-background-transition="slide">
<h4 id="orgc511537"></h4>
</section>
<section id="slide-orgb7849c2">
<h4 id="orgb7849c2">Gene Centric vs. Enhancer Centric</h4>
<ul>
<li>Peng&rsquo;s approach
<ul>
<li>Took enhancers that were expressed deferentially</li>
<li>Linked them to Genes within 200Kb</li>

</ul></li>
<li>New approach
<ul>
<li>Find genes that are deferentially expressed</li>
<li>Link the Enhancers to those genes</li>

</ul></li>

</ul>

</section>
<section id="slide-org760edd3" data-background="viz/pipeline.png" data-background-size="600px" data-background-transition="slide">
<h4 id="org760edd3"></h4>
</section>
<section id="slide-org684901d">
<h4 id="org684901d">featureCounts</h4>
<div class="outline-text-4" id="text-org684901d">
</div>
<ul class="org-ul">
<li><a id="org5b0d100"></a>Tried CLI version<br /></li>
<li><a id="org43cf30e"></a>Couldn&rsquo;t get examples to work with edgeR<br /></li>
<li><a id="org38e694b"></a>Switched to R package<br /></li>
<li><a id="org578a036"></a>Switched to a snakemake wrapper<br /></li>
</ul>
</section>
<section id="slide-org634e912">
<h4 id="org634e912">Limma</h4>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a></li>

</ul>
</section>
<section id="slide-orgf346dd9" data-background="img/limma/GM19_fig3_limma.png" data-background-size="800px" data-background-transition="slide">
<h4 id="orgf346dd9"></h4>
</section>
<section id="slide-orgbc1e2bd" data-background="img/limma/IMR_fig3_limma.png" data-background-size="800px" data-background-transition="slide">
<h4 id="orgbc1e2bd"></h4>
</section>
</section>
<section>
<section id="slide-orgace6da4">
<h3 id="orgace6da4">Overview</h3>
<ul>
<li>Get List of inducible Genes</li>
<li>Link inducible eRNAs and inducible genes</li>
<li>Checking for L2</li>
<li>Compare IMR and GM inducible lists</li>

</ul>
</section>
<section id="slide-org0cc0664">
<h4 id="org0cc0664">Get List of DE Genes</h4>
<ul>
<li>Used TPM values</li>
<li>Removed Genes without a foldchange &gt; 1 between any timepoints</li>
<li>Considering using only points t&lt;24h for inducibility</li>
<li>Working to implement one-step autocorrelation</li>

</ul>
</section>
<section id="slide-orgb6ed140">
<h4 id="orgb6ed140">Link eRNAs and Top genes</h4>
<ul>
<li>Awk script to get RefSeq, with only specific BED entries</li>
<li>Ended up using Ripgrep</li>

</ul>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>awk -F <span style="color: #98be65;">\"\\</span>t<span style="color: #98be65;">\"</span> <span style="color: #98be65;">'{{ if (NR!=1){{ print $1 }}}}'</span> genesFoldChange.txt | <span style="color: #98be65;">\</span>
rg --dfa-size-limit 2G -w -f - refseqgenes.gtf | <span style="color: #98be65;">\</span>
rg -w exon - &gt; de_genes.gtf
</code></pre>
</div>
<ul>
<li>Linked eRNAs within 200Kb</li>

</ul>
</section>
<section id="slide-org8d292e4">
<h4 id="org8d292e4">Compare IMR and GM eRNAs</h4>
<ul>
<li>Created Meta eRNAs
<ul>
<li>Find overlaps of IMR and GM (Viral)</li>
<li>Merge GM and IMR eRNAs(Meta)</li>
<li>Report Meta Viral eRNAs</li>

</ul></li>

</ul>
</section>
<section id="slide-org42bd820" data-background="img/venn/eRNA_cross_cell.svg" data-background-size="1000px" data-background-transition="slide">
<h4 id="org42bd820"></h4>
</section>
<section id="slide-org9188360" data-background="img/venn/eRNA_cross_cell_viral.svg" data-background-size="1000px" data-background-transition="slide">
<h4 id="org9188360"></h4>
</section>
<section id="slide-orge700dae">
<h4 id="orge700dae">Checking for L2</h4>
</section>
<section id="slide-orge138bd7" data-background="img/L2_GenomeBrowser_v2.png" data-background-size="1000px" data-background-transition="slide">
<h4 id="orge138bd7"></h4>
</section>
<section id="slide-org94d761f" data-background="img/peaks.groseqChart.png" data-background-size="1000px" data-background-transition="slide">
<h4 id="org94d761f"></h4>
</section>
<section id="slide-orga0d4403">
<h4 id="orga0d4403">Possible Groupings</h4>
</section>
<section id="slide-orga568afc" data-background="img/grouping_eRNAs.png" data-background-size="1500px" data-background-transition="slide">
<h4 id="orga568afc"></h4>
</section>
</section>
<section>
<section id="slide-orgc83657e">
<h3 id="orgc83657e"><a href="https://emiller88.github.io/presentations/Functional_Genomics/2019-10-16/report.html">Snakemake Report</a></h3>
</section>
</section>
<section>
<section id="slide-org5d5826c">
<h3 id="org5d5826c">Future</h3>
<ul>
<li>Clean Groupings</li>
<li>Tweak Homer</li>
<li>Work with TNFSF10 to test grouping Identification Pipeline</li>

</ul>
</section>
</section>
</div>
</div>
<script src="https://cdn.jsdelivr.net/npm/reveal.js/js/reveal.js"></script>

<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({
multiplex: {
    secret: '', // null if client
    id: '', // id, obtained from socket.io server
    url: '' // Location of socket.io server
},

// Optional libraries used to extend on reveal.js
dependencies: [
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }]
});
</script>
</body>
</html>
