<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Xonsh, Nix, &amp; Limma</title>
    <meta name="author" content="(Edmund Miller)" />
    <style type="text/css">
      .underline {
        text-decoration: underline;
      }
    </style>
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/reveal.js/css/reveal.css"
    />

    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/reveal.js/css/theme/white.css"
      id="theme"
    />

    <!-- If the query includes 'print-pdf', include the PDF print sheet -->
    <script>
      if (window.location.search.match(/print-pdf/gi)) {
        var link = document.createElement("link");
        link.rel = "stylesheet";
        link.type = "text/css";
        link.href = "https://cdn.jsdelivr.net/npm/reveal.js/css/print/pdf.css";
        document.getElementsByTagName("head")[0].appendChild(link);
      }
    </script>
  </head>
  <body>
    <div class="reveal">
      <div class="slides">
        <section>
          <section id="slide-orgb879949">
            <h2 id="orgb879949">Xonsh</h2>
            <div class="outline-text-2" id="text-orgb879949"></div>
          </section>
          <section id="slide-orgb1eea22">
            <h3 id="orgb1eea22">SciPy 2019</h3>
            <div class="outline-text-3" id="text-orgb1eea22"></div>
          </section>
          <section id="slide-orgd5f7a0b">
            <h4 id="orgd5f7a0b">Links</h4>
            <div class="outline-text-4" id="text-orgd5f7a0b"></div>
            <ul class="org-ul">
              <li>
                <a id="org5ec3c8f"></a
                ><a href="http://conference.scipy.org/">Conference List</a
                ><br />
              </li>
              <li>
                <a id="org1448870"></a
                ><a href="https://xonsh.github.io/scipy-2019-tutorial/#1"
                  >Xonsh Tutorial</a
                ><br />
              </li>
              <li>
                <a id="orgd50344a"></a
                ><a
                  href="https://youtu.be/ujo9sbqHFqw?list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC"
                  >Xonsh Talk</a
                ><br />
              </li>
              <li>
                <a id="org82430c1"></a
                ><a
                  href="https://www.youtube.com/watch?v=ZB7BZMhfPgk&amp;list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC"
                  >Talk Playlist</a
                ><br />
              </li>
            </ul>
          </section>
        </section>
        <section>
          <section id="slide-org35754b8">
            <h2 id="org35754b8">Nix</h2>
            <div class="outline-text-2" id="text-org35754b8"></div>
          </section>
          <section id="slide-org61410fc">
            <h3 id="org61410fc">Code as Infastructure</h3>
            <ul>
              <li>
                <a href="https://nixos.org/~eelco/pubs/nixos-jfp-final.pdf"
                  >NixOS: A Purely Functional Linux Distribution</a
                >
              </li>
            </ul>
          </section>
          <section id="slide-orgfd76c53">
            <h3 id="orgfd76c53">A Couple of Different Things</h3>
            <ul>
              <li>Nix Programming Language</li>
              <li>NIx Package Manager</li>
              <li>NixOS</li>
              <li>
                <a
                  href="https://www.youtube.com/watch?v=USDbjmxEZ_I&amp;list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC&amp;index=16"
                  >SciPy Talk</a
                >
              </li>
            </ul>
          </section>
          <section
            id="slide-org35d79e5"
            data-background="img/HowNix.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h3 id="org35d79e5"></h3>
          </section>
          <section id="slide-orgc2a0a55">
            <h3 id="orgc2a0a55">Simple Derivation</h3>
            <div class="org-src-container">
              <pre
                class="src src-sh"
              ><code trim><code><span style="color: #ff6ac1;">declare</span> -xp</code>
<code><span style="color: #ff6ac1;">export</span> <span style="color: #ff5c57;">PATH</span>=<span style="color: #f3f99d;">"</span><span style="color: #5af78e;">$</span><span style="color: #ff5c57;">coreutils</span><span style="color: #f3f99d;">/bin"</span></code>
<code><span style="color: #9aedfe;">echo</span> <span style="color: #f3f99d;">"hello"</span></code>
<code><span style="color: #9aedfe;">echo</span> $<span style="color: #ff5c57;">blah</span></code>
<code><span style="color: #ff5c57;">dest</span>=$<span style="color: #ff5c57;">out</span>/bin</code>
<code><span style="color: #9aedfe;">mkdir</span> -p $<span style="color: #ff5c57;">dest</span></code>
<code><span style="color: #9aedfe;">cp</span> $<span style="color: #ff5c57;">src</span> $<span style="color: #ff5c57;">dest</span>/my-hello</code>
</code></pre>
            </div>
          </section>
          <section id="slide-org1d9cb35">
            <h3 id="org1d9cb35">Simple Derivation</h3>
            <div class="org-src-container">
              <pre
                class="src src-nix"
              ><code trim><code><span style="color: #ffb86c;">with</span> (<span style="color: #ff6ac1;">import</span> <span style="color: #5af78e;">&lt;nixpkgs&gt;</span> {});</code>
<code><span style="color: #ff6ac1;">derivation</span> {</code>
<code>  <span style="color: #ff5c57;">name</span> = <span style="color: #f3f99d;">"my_hello"</span>;</code>
<code>  <span style="color: #ff5c57;">builder</span> = <span style="color: #f3f99d;">"</span><span style="color: #ff6ac1; font-weight: bold;">${</span>bash<span style="color: #ff6ac1; font-weight: bold;">}</span><span style="color: #f3f99d;">/bin/bash"</span>;</code>
<code>  <span style="color: #ff5c57;">args</span> = [ <span style="color: #5af78e;">./build.sh</span> ];</code>
<code>  <span style="color: #ff5c57;">system</span> = builtins.currentSystem;</code>
<code>  <span style="color: #ff5c57;">coreutils</span> = coreutils;</code>
<code>  <span style="color: #ff5c57;">src</span> = <span style="color: #5af78e;">./my-hello</span>;</code>
<code>  <span style="color: #ff5c57;">blah</span> = <span style="color: #5af78e;">./blah</span>;</code>
<code>}</code>
</code></pre>
            </div>
          </section>
          <section id="slide-org8d3eac3">
            <h3 id="org8d3eac3">My Config</h3>
          </section>
          <section id="slide-orgb1b70ad">
            <h3 id="orgb1b70ad">Python and Nix</h3>
            <div class="outline-text-3" id="text-orgb1b70ad"></div>
          </section>
          <section id="slide-orgdffeea8">
            <h4 id="orgdffeea8">Imperative Installation</h4>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><code trim><code>$ nix-env -i hello</code>
<code>$ nix-env --list-generations</code>
<code>$ nix-env --rollback</code>
<code><span style="color: #606580;"># </span><span style="color: #606580;">View environment dependencies</span></code>
<code>$ nix-env -q</code>
</code></pre>
            </div>
          </section>
          <section id="slide-orge776a67">
            <h4 id="orge776a67">nix-shell</h4>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><code trim><code>$ nix-shell --pure -p python27 -p python27Packages.numpy</code>
</code></pre>
            </div>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><code trim><code>$ python -c <span style="color: #f3f99d;">"import numpy"</span></code>
<code>$ <span style="color: #9aedfe;">echo</span> $<span style="color: #ff5c57;">PYTHONPATH</span></code>
</code></pre>
            </div>
          </section>
          <section id="slide-orga373c6c">
            <h4 id="orga373c6c">Repeatable Environment</h4>
            <div class="org-src-container">
              <pre
                class="src src-nix"
              ><code trim><code><span style="color: #606580;"># </span><span style="color: #606580;">shell.nix</span></code>
<code><span style="color: #ffb86c;">let</span></code>
<code>  <span style="color: #ff5c57;">nixpkgs</span> = <span style="color: #ff6ac1;">import</span> <span style="color: #5af78e;">&lt;nixpkgs&gt;</span> {};</code>
<code>  <span style="color: #ff5c57;">pypkgs</span> = nixpks.python36Packages;</code>
<code><span style="color: #ffb86c;">in</span></code>
<code>  (pypkgs.python.withPackages (x: [ x.numpy x.scipy])).env</code>
</code></pre>
            </div>
            <ul>
              <li>Can also prevent version drift</li>
            </ul>
          </section>
          <section id="slide-orgd8ee3ee">
            <h4 id="orgd8ee3ee">buildPythonPackage</h4>
            <div class="org-src-container">
              <pre
                class="src src-nix"
              ><code trim><code>{ nixpkgs, pypkgs }:</code>
<code>pypkgs.buildPythonPackage <span style="color: #ffb86c;">rec</span> {</code>
<code>  <span style="color: #ff5c57;">pname</span> = <span style="color: #f3f99d;">"scipy"</span>;</code>
<code>  <span style="color: #ff5c57;">version</span> = <span style="color: #f3f99d;">"0.19.1"</span>;</code>
<code>  <span style="color: #ff5c57;">src</span> = pypkgs.fetchPypi {</code>
<code>    <span style="color: #ffb86c;">inherit</span> pname version;</code>
<code>    <span style="color: #ff5c57;">sha256</span> = <span style="color: #f3f99d;">"1rl411bvla6q7qfdb47fpdnyjhfgzl6smpha33n9ar1klykjr6m1"</span>;</code>
<code>  };</code>
<code>  <span style="color: #ff5c57;">buildInputs</span> = [</code>
<code>    pypkgs.numpy</code>
<code>    nixpkgs.pkgs.gcc</code>
<code>    nixpkgs.pkgs.gfortran</code>
<code>  ];</code>
<code>  <span style="color: #ff5c57;">ignoreCollisions</span>=true;</code>
<code>  <span style="color: #ff5c57;">catchConflicts</span>=false;</code>
<code>  <span style="color: #ff5c57;">doCheck</span> = <span style="color: #ff6ac1;">false</span>;</code>
<code>}</code>
</code></pre>
            </div>
            <ul>
              <li>Fetch from PyPi</li>
              <li>Specify build time, run time, tests, dependancies</li>
              <li>
                Runs <code>setup.py install</code> or
                <code>setup.py develop</code>
              </li>
            </ul>
          </section>
          <section id="slide-org8ecfec1">
            <h4 id="org8ecfec1">PyPi2Nix</h4>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><code trim><code>$ pypi2nix -V <span style="color: #f3f99d;">"3.6"</span> -r requirements.txt</code>
<code>$ nix-shell --pure requirements.nix</code>
</code></pre>
            </div>
            <ul>
              <li>Doesn't matter if packages aren't in nix packages</li>
            </ul>
          </section>
        </section>
        <section>
          <section id="slide-orgf5ca54f">
            <h2 id="orgf5ca54f">
              eRNA Prediction Pipeline from GRO-Seq INFB1 Induction Timecourse
            </h2>
            <div class="outline-text-2" id="text-orgf5ca54f"></div>
          </section>
          <section id="slide-orgfe74187">
            <h3 id="orgfe74187">Recap</h3>
            <div class="outline-text-3" id="text-orgfe74187"></div>
          </section>
          <section
            id="slide-org67839d1"
            data-background="img/globaltrans.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org67839d1"></h4>
          </section>
          <section id="slide-org9c7d925">
            <h4 id="org9c7d925">Overview</h4>
            <ul>
              <li>Reproducing GM18</li>
              <li>Predicted IMR90 eRNAs</li>
              <li>Compared IMR90 Predicted Enhancers to GM</li>
              <li>Used Homer scripts to find DE of eRNAs and Genes</li>
              <li>Gene Centric vs. Enhancer Centric</li>
            </ul>
          </section>
          <section id="slide-orge40e647">
            <h4 id="orge40e647">Reproducing GM18</h4>
            <ul>
              <li>hg18 vs hg19</li>
              <li>Overpredicting eRNA transcripts</li>
              <li>
                Past Issue
                <ul>
                  <li>What I thought Peng sent me</li>
                  <li>hg18 -&gt; eRNAs -&gt; Me</li>
                  <li>What actually happened</li>
                  <li>hg18 -&gt; eRNAs -&gt; LiftOver -&gt; hg19 -&gt; Me</li>
                </ul>
              </li>
              <li>Main issue is homer uniqmap</li>
            </ul>
          </section>
          <section
            id="slide-org1b33a43"
            data-background="img/Figure_1.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org1b33a43"></h4>
          </section>
          <section
            id="slide-orgda723a9"
            data-background="img/Figure_2.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgda723a9"></h4>
          </section>
          <section
            id="slide-org6ac1191"
            data-background="img/dag.png"
            data-background-size="400px"
            data-background-transition="slide"
          >
            <h4 id="org6ac1191"></h4>
          </section>
          <section id="slide-orga69dc48">
            <h4 id="orga69dc48">Actual Pipelines</h4>
            <div class="outline-text-4" id="text-orga69dc48"></div>
            <ul class="org-ul">
              <li>
                <a id="orga6fac74"></a><a href="files/GM18dag.pdf">GM18</a
                ><br />
              </li>
              <li>
                <a id="org6df6c55"></a><a href="files/IMR19dag.pdf">IMR19</a
                ><br />
              </li>
              <li>
                <a id="org1fc0945"></a><a href="files/all.pdf">All</a><br />
              </li>
            </ul>
          </section>
          <section id="slide-orgd843425">
            <h4 id="orgd843425">Predicted IMR90 eRNAs</h4>
            <p>
              Changes from GM18
            </p>
            <ul>
              <li>hg19</li>
              <li>No liftover</li>
            </ul>
          </section>
          <section id="slide-org7024833">
            <h4 id="org7024833">Compared IMR90 Predicted Enhancers to GM</h4>
          </section>
          <section
            id="slide-org7581dc1"
            data-background="img/Venn.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org7581dc1"></h4>
          </section>
          <section id="slide-orgec0a528">
            <h4 id="orgec0a528">
              Used Homer scripts to find DE of eRNAs and Genes
            </h4>
          </section>
          <section id="slide-orgeea02f4">
            <h4 id="orgeea02f4">Gene Centric vs. Enhancer Centric</h4>
            <ul>
              <li>
                Peng's approach
                <ul>
                  <li>Took enhancers that were expressed deferentially</li>
                  <li>Linked them to Genes within 200Kb</li>
                </ul>
              </li>
              <li>
                New approach
                <ul>
                  <li>Take genes that are deferentially expressed</li>
                  <li>Link the Enhancers to those genes</li>
                </ul>
              </li>
            </ul>
          </section>
          <section
            id="slide-orgeb306b3"
            data-background="viz/pipeline.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgeb306b3"></h4>
          </section>
          <section id="slide-orgdd975f8">
            <h4 id="orgdd975f8">featureCounts</h4>
            <div class="outline-text-4" id="text-orgdd975f8"></div>
            <ul class="org-ul">
              <li><a id="org8b3ab8b"></a>Tried CLI version<br /></li>
              <li>
                <a id="orgc48688f"></a>Couldn't get examples to work with
                edgeR<br />
              </li>
              <li><a id="org04e9982"></a>Switched to R package<br /></li>
              <li><a id="org7ed8e45"></a>Update: switched back<br /></li>
            </ul>
          </section>
          <section id="slide-orga472083">
            <h3 id="orga472083">New Things</h3>
            <div class="outline-text-3" id="text-orga472083"></div>
          </section>
          <section id="slide-org8d4b9cf">
            <h4 id="org8d4b9cf">Shayne Easterwood</h4>
            <ul>
              <li>NOISeq Analysis</li>
              <li>SAF Format for eRNA formats</li>
              <li>A chance to clean up</li>
            </ul>
          </section>
          <section id="slide-org1d30387">
            <h4 id="org1d30387">Linear Model / Time Series Analysis</h4>
          </section>
          <section id="slide-orge464b80">
            <h4 id="orge464b80">Limma</h4>
            <ul>
              <li>
                <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/"
                  >RNA-seq analysis is easy as 1-2-3 with limma, Glimma and
                  edgeR</a
                >
              </li>
              <li>
                <a
                  href="http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MDS-Plot.html"
                  >glimma Plots</a
                >
              </li>
            </ul>
          </section>
          <section
            id="slide-org15bc123"
            data-background="img/limma/GM19_fig1_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org15bc123"></h4>
          </section>
          <section
            id="slide-org08f6ead"
            data-background="img/limma/IMR_fig1_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org08f6ead"></h4>
          </section>
          <section
            id="slide-orgdc5c025"
            data-background="img/limma/GM19_fig2_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgdc5c025"></h4>
          </section>
          <section
            id="slide-orgdb30c80"
            data-background="img/limma/IMR_fig2_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgdb30c80"></h4>
          </section>
          <section
            id="slide-orga26bc8a"
            data-background="img/limma/GM19_fig3_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orga26bc8a"></h4>
          </section>
          <section
            id="slide-orge14ca1e"
            data-background="img/limma/IMR_fig3_limma.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orge14ca1e"></h4>
          </section>
          <section id="slide-org6076838">
            <h4 id="org6076838">Future</h4>
            <ul>
              <li>Get List of Top Genes</li>
              <li>Link eRNAs and Top genes</li>
              <li>Compare IMR and GM Genes</li>
              <li>Kinetics of the eRNAs</li>
            </ul>
          </section>
        </section>
      </div>
    </div>
    <script src="https://cdn.jsdelivr.net/npm/reveal.js/js/reveal.js"></script>

    <script>
      // Full list of configuration options available here:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        controls: true,
        progress: true,
        history: false,
        center: true,
        slideNumber: "c",
        rollingLinks: false,
        keyboard: true,
        overview: true,

        theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
        transition: Reveal.getQueryHash().transition || "default", // default/cube/page/concave/zoom/linear/fade/none
        transitionSpeed: "default",
        multiplex: {
          secret: "", // null if client
          id: "", // id, obtained from socket.io server
          url: "" // Location of socket.io server
        },

        // Optional libraries used to extend on reveal.js
        dependencies: [
          ,
          {
            src: "https://cdn.jsdelivr.net/npm/reveal.js/lib/js/classList.js",
            condition: function() {
              return !document.body.classList;
            }
          },
          {
            src:
              "https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/marked.js",
            condition: function() {
              return !!document.querySelector("[data-markdown]");
            }
          },
          {
            src:
              "https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/markdown.js",
            condition: function() {
              return !!document.querySelector("[data-markdown]");
            }
          },
          {
            src:
              "https://cdn.jsdelivr.net/npm/reveal.js/plugin/zoom-js/zoom.js",
            async: true,
            condition: function() {
              return !!document.body.classList;
            }
          },
          {
            src: "https://cdn.jsdelivr.net/npm/reveal.js/plugin/notes/notes.js",
            async: true,
            condition: function() {
              return !!document.body.classList;
            }
          }
        ]
      });
    </script>
  </body>
</html>
