<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>nf-core &amp; Selene</title>
<meta name="author" content="Edmund Miller"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/reveal.css"/>

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/theme/white.css" id="theme"/>

<!-- If the query includes 'print-pdf', include the PDF print sheet -->
<script>
    if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'https://cdn.jsdelivr.net/npm/reveal.js/css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
</script>
</head>
<body>
<div class="reveal">
<div class="slides">
<section>
<section id="slide-org09e0c0b">
<h2 id="org09e0c0b"><object type="image/svg+xml" data="https://nf-co.re/assets/img/logo/nf-core-logo.svg" class="org-svg">
Sorry, your browser does not support SVG.</object></h2>
<p>
<a href="https://nf-co.re/">https://nf-co.re/</a>
</p>
<blockquote>
<p>
A community effort to collect a curated set of analysis pipelines built using Nextflow.
</p>
</blockquote>
<p>
<a href="https://www.nature.com/articles/s41587-020-0439-x.epdf?shared_access_token=W-gHNzA4HngzDDSTRrfBRdRgN0jAjWel9jnR3ZoTv0P9zw2JZTis0qRZaQ0qtypN0g4rwhNebqVD1V-gt-AOlA82ul659wBl7hYkgj3IQbVny9QNsU0jdhfynPtglNcfpqimaGrEVJmM7gUo8H6lyg%3D%3D">Full Access Link</a>
</p>
</section>
</section>
<section>
<section id="slide-org393eb03">
<h3 id="org393eb03">Where we left off</h3>
<div class="org-src-container">

<pre  class="src src-shell">cookiecutter gl:functional-genomics/cookiecutter-snakemake-workflow
</pre>
</div>
</section>
</section>
<section>
<section id="slide-orgd2eab46" data-background="https://images.readcube-cdn.com/publishers/nature/figures/e7b002cb315261aa851aa616874ad980e20a6ad86a47854610d2853fca66adbc/1.jpg" data-background-size="800px" data-background-transition="slide">
<h3 id="orgd2eab46"></h3>
</section>
</section>
<section>
<section id="slide-orgfdaa3da">
<h3 id="orgfdaa3da">Deploy</h3>
<div class="outline-text-3" id="text-orgfdaa3da">
</div>
</section>
<section id="slide-org4f1f1d5">
<h4 id="org4f1f1d5">Getting started</h4>
<div class="org-src-container">

<pre  class="src src-sh"><span style="color: #5B6268;"># </span><span style="color: #5B6268;">Install Nextflow</span>
<span style="color: #ECBE7B;">curl</span> -fsSL get.nextflow.io | bash
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Add Nextflow binary to your user's PATH:</span>
<span style="color: #ECBE7B;">mv</span> nextflow ~/bin/

nextflow run nf-core/rnaseq -profile docker <span style="color: #98be65;">\</span>
    --singleEnd --reads <span style="color: #98be65;">'*.fastq'</span> --genome GRCm38

nextflow run nf-core/rnaseq -profile utd_ganymede <span style="color: #98be65;">\</span>
    --singleEnd --reads <span style="color: #98be65;">'*.fastq'</span> --genome GRCm38
</pre>
</div>
</section>
<section id="slide-org57909f7">
<h4 id="org57909f7"><a href="https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc">Demo</a></h4>
<a href="https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc" target="_blank"><img src="https://asciinema.org/a/EfVHOhIS3XC3nzWm0NwupLgGc.svg" height="550"/></a>
</section>
<section id="slide-org163f2dc">
<h4 id="org163f2dc">MK config coming soon!</h4>

</section>
</section>
<section>
<section id="slide-org8360ebf">
<h3 id="org8360ebf">Participate</h3>
<div class="outline-text-3" id="text-org8360ebf">
</div>
</section>
<section id="slide-orgbcbe5c9">
<h4 id="orgbcbe5c9">Documentation</h4>
</section>
<section id="slide-org9b47df8">
<h4 id="org9b47df8"><a href="https://nf-co.re/pipelines">Pipelines</a></h4>
<ul>
<li><a href="https://nf-co.re/rnaseq">RNA-Seq</a></li>
<li><a href="https://nf-co.re/scrnaseq">scRNA-Seq</a></li>
<li><a href="https://nf-co.re/atacseq">ATAC-Seq</a></li>
<li><a href="https://nf-co.re/deepvariant">Deep Variant</a></li>

</ul>
</section>
<section id="slide-org8a1ef32">
<h4 id="org8a1ef32">RNA-Seq</h4>
<p>
<a href="https://nf-co.re/rnaseq/docs/output#pipeline-overview">pipeline-overview</a>
</p>
<ul>
<li>fastqc/TrimGalore</li>
<li>STAR/hisat/salmon</li>
<li>featureCounts</li>
<li>edgeR</li>
<li>Multiqc</li>

</ul>
</section>
<section id="slide-org0b84996">
<h4 id="org0b84996">Forking</h4>
</section>
</section>
<section>
<section id="slide-orgff2e673">
<h3 id="orgff2e673">Develop</h3>
<div class="outline-text-3" id="text-orgff2e673">
</div>
</section>
<section id="slide-org33beca2">
<h4 id="org33beca2"><a href="https://nf-co.re/tools#creating-a-new-workflow">Creating a new workflow</a></h4>
<div class="org-src-container">

<pre  class="src src-sh">pip install nf-core
nf-core create
</pre>
</div>
</section>
<section id="slide-orgfdda73f">
<h4 id="orgfdda73f"><a href="https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx">Demo</a></h4>
<a href="https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx" target="_blank"><img src="https://asciinema.org/a/8VfGOnzYKPhK6nGubG9pBkHcx.svg" height="550"/></a>
</section>
</section>
<section>
<section id="slide-org2cbe7ae">
<h2 id="org2cbe7ae"><img src="https://github.com/FunctionLab/selene/raw/master/docs/source/_static/img/selene_logo.png" alt="selene_logo.png" /></h2>
<div class="outline-text-2" id="text-org2cbe7ae">
</div>
</section>
</section>
<section>
<section id="slide-orgc0bd7ea">
<h3 id="orgc0bd7ea">Links</h3>
<ul>
<li><a href="https://www.nature.com/articles/s41592-019-0360-8">Selene: a PyTorch-based deep learning library for sequence data | Nature Methods</a></li>
<li><a href="https://github.com/FunctionLab/selene">GitHub - FunctionLab/selene</a></li>

</ul>
</section>
</section>
<section>
<section id="slide-org5595234">
<h3 id="org5595234"><a href="https://hb.flatironinstitute.org/deepsea/">DeepSEA</a></h3>
<p>
DeepSEA is a deep learning-based algorithmic framework for predicting the
chromatin effects of sequence alterations with single nucleotide sensitivity.
</p>
</section>
</section>
<section>
<section id="slide-orga650c60">
<h3 id="orga650c60">Overview</h3>
<ul>
<li>Trying to solve the issue of creating new deep learning architectures and
training them on new data sets.</li>
<li>Created to be a framework to be easy to use for deep learning in biology.</li>

</ul>
</section>
</section>
<section>
<section id="slide-orga76fcf8">
<h3 id="orga76fcf8">Case 1: Training an exisiting architecture on a different dataset</h3>
<ul>
<li>Trains DeepSEA module for transcription factor GATA1</li>
<li>Zero lines of python code</li>

</ul>
</section>
</section>
<section>
<section id="slide-org7e923e0">
<h3 id="org7e923e0">Case 2: Developing a new architecture and comparing performance across architectures</h3>
<ul>
<li>Allows use of PyTorch modules</li>
<li>Fully compatible with <a href="https://kipoi.org/">Kipoi</a></li>
<li>Focus is on experimenting with various models rather than repeating code for
training and evaluation.</li>

</ul>
</section>
</section>
<section>
<section id="slide-org3b107aa">
<h3 id="org3b107aa">Case 3: Applying a new model to variants</h3>
<ul>
<li>They train the new model from Case 2 to be used on the Alzheimer&rsquo;s disease
GWAS to find varent effects that are significant and non-significant.</li>

</ul>
</section>
</section>
<section>
<section id="slide-org607e41a">
<h3 id="org607e41a">Online methods</h3>
<ul>
<li>Has an SDK and a CLI</li>
<li>SDK is a python package for use in example with a web app to allow researchers
to submit sequences.</li>

</ul>
</section>
</section>
<section>
<section id="slide-org62ac00b">
<h2 id="org62ac00b">Learning Resources</h2>
<ul>
<li><a href="https://www.coursera.org/learn/sciwrite">Writing in the Science</a></li>
<li><a href="https://missing.csail.mit.edu/">The Missing Semester</a></li>
<li><a href="https://takesmartnotes.com/">How to Take Smart Notes</a></li>

</ul>
</section>
</section>
</div>
</div>
<script src="https://cdn.jsdelivr.net/npm/reveal.js/lib/js/head.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/reveal.js/js/reveal.js"></script>
<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({

controls: true,
progress: true,
history: false,
center: true,
slideNumber: 'c',
rollingLinks: false,
keyboard: true,
mouseWheel: false,
fragmentInURL: false,
hashOneBasedIndex: false,
pdfSeparateFragments: true,

overview: true,

theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
transition: Reveal.getQueryHash().transition || 'convex', // see README of reveal.js for options
transitionSpeed: 'default',

// Optional libraries used to extend reveal.js
dependencies: [
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/search/search.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } }]

});
</script>
</body>
</html>
