<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Intro</title>
    <meta name="author" content="(Edmund Miller)" />
    <style type="text/css">
      .underline {
        text-decoration: underline;
      }
    </style>
    <link
      rel="stylesheet"
      href="http://cdn.jsdelivr.net/reveal.js/3.0.0/css/reveal.css"
    />

    <link
      rel="stylesheet"
      href="http://cdn.jsdelivr.net/reveal.js/3.0.0/css/theme/white.css"
      id="theme"
    />

    <!-- If the query includes 'print-pdf', include the PDF print sheet -->
    <script>
      if (window.location.search.match(/print-pdf/gi)) {
        var link = document.createElement("link");
        link.rel = "stylesheet";
        link.type = "text/css";
        link.href = "http://cdn.jsdelivr.net/reveal.js/3.0.0/css/print/pdf.css";
        document.getElementsByTagName("head")[0].appendChild(link);
      }
    </script>
  </head>
  <body>
    <div class="reveal">
      <div class="slides">
        <section>
          <section id="slide-org40b1373">
            <h2 id="org40b1373">eRNA Kinetics from INFB1 induction</h2>
          </section>
        </section>
        <section>
          <section id="slide-org96b2cbc">
            <h2 id="org96b2cbc">Who am I</h2>
            <ul>
              <li>Plano, TX</li>
              <li><a href="https://github.com/Emiller88">GitHub</a></li>
              <li>Undergraduate Molecular Biology '15-'18</li>
              <li>Masters Biotechnology '18-'19</li>
              <li>Joined Functional Genomics group in July '15</li>
              <li>Moved to Dry Lab Summer '17</li>
              <li>Software Developer for Olypsis Technologies since May '18</li>
            </ul>
          </section>
          <section id="slide-org28c4253">
            <h3 id="org28c4253">What I am looking to get out of this</h3>
          </section>
          <section
            id="slide-org2ef68c8"
            data-background="img/biovenn.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h3 id="org2ef68c8"></h3>
          </section>
          <section id="slide-orgcd57a95">
            <h3 id="orgcd57a95">Things I can bring to the table</h3>
            <ul>
              <li>Wet Lab</li>
              <li>Linux</li>
              <li>Emacs</li>
              <li>Git</li>
              <li>Devops &amp; CI/CD</li>
              <li>Docker</li>
              <li>Snakemake</li>
            </ul>
          </section>
        </section>
        <section>
          <section id="slide-org67d3cdc">
            <h2 id="org67d3cdc">Reproducible Research</h2>
            <div class="outline-text-2" id="text-org67d3cdc"></div>
          </section>
          <section id="slide-org7b7fa40">
            <h3 id="org7b7fa40">Creating a Workflow</h3>
            <div class="outline-text-3" id="text-org7b7fa40"></div>
          </section>
          <section id="slide-orgc35d308">
            <h4 id="orgc35d308">Emacs</h4>
            <ul>
              <li>Evil Keybinding</li>
              <li>Magit</li>
              <li>Dired</li>
              <li>Tramp</li>
              <li>Org-mode</li>
              <li>Email</li>
              <li>RSS</li>
            </ul>
          </section>
          <section id="slide-org54eab4f">
            <h4 id="org54eab4f">Spacemacs</h4>
            <ul>
              <li>Slow</li>
              <li>Difficult to Customize outside of the box</li>
            </ul>
          </section>
          <section id="slide-orgcbaa815">
            <h4 id="orgcbaa815">Doom Emacs</h4>
            <ul>
              <li>Optimized</li>
              <li>Community</li>
            </ul>
          </section>
          <section id="slide-orgcf0e1f7">
            <h4 id="orgcf0e1f7">Org-Mode</h4>
            <ul>
              <li>Slides</li>
              <li>TODOs</li>
              <li>Agenda</li>
              <li>Capture</li>
              <li>Lab-Notebook</li>
              <li>Babel</li>
            </ul>
          </section>
          <section id="slide-orgea44df0">
            <h3 id="orgea44df0">Makefiles</h3>
            <div class="outline-text-3" id="text-orgea44df0"></div>
          </section>
          <section
            id="slide-orge574daf"
            data-background="img/data.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orge574daf"></h4>
          </section>
          <section
            id="slide-orgeeb4c50"
            data-background="img/data1.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgeeb4c50"></h4>
          </section>
          <section
            id="slide-orgb24548d"
            data-background="img/data2.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="orgb24548d"></h4>
          </section>
          <section id="slide-org4c040fb">
            <h4 id="org4c040fb">Example Makefile</h4>
            <div class="org-src-container">
              <pre
                class="src src-makefile"
              ><span style="color: #51afef;">all</span>: prot001.txt prot0002.txt prot003.txt

<span style="color: #51afef;">%.txt</span>: ./bin/intensive_operation %.fasta
   $<span style="color: #a9a1e1;">^</span> &gt; <span style="color: #51afef;">$</span><span style="color: #a9a1e1;">@</span>

<span style="color: #51afef;">%.fasta</span>:
    curl database.example.com/<span style="color: #51afef;">$</span><span style="color: #a9a1e1;">*</span> &gt; <span style="color: #51afef;">$</span><span style="color: #a9a1e1;">@</span>
</pre>
            </div>
          </section>
          <section id="slide-orgfd9b895">
            <h3 id="orgfd9b895">Bioinformatics Pipelines</h3>
            <div class="outline-text-3" id="text-orgfd9b895"></div>
          </section>
          <section id="slide-org9705381">
            <h4 id="org9705381">Toil</h4>
            <ul>
              <li>Python 2.7</li>
              <li>UCSC</li>
              <li>Complex</li>
            </ul>
          </section>
          <section id="slide-org912356e">
            <h4 id="org912356e">Nextflow</h4>
            <ul>
              <li>Java</li>
              <li>Barcelona Center for Genomic Regulation</li>
              <li>Push</li>
              <li>Able to script in any language</li>
            </ul>
          </section>
          <section id="slide-org59b4078">
            <h4 id="org59b4078">Snakemake</h4>
            <ul>
              <li>Python 3.5</li>
              <li>Easy to containerize</li>
              <li>Pull</li>
              <li>Quick to use</li>
            </ul>
          </section>
          <section
            id="slide-orgcffcea3"
            data-background="img/snakemake.png"
            data-background-size="1000px"
            data-background-transition="slide"
          >
            <h4 id="orgcffcea3"></h4>
          </section>
          <section id="slide-org716c48a">
            <h4 id="org716c48a">Snakemake Example</h4>
            <div class="org-src-container">
              <pre
                class="src src-snakemake"
              ><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">targets</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"plots/dataset1.pdf"</span>,
        <span style="color: #98be65;">"plots/dataset2.pdf"</span>

<span style="color: #51afef;">rule</span> <span style="color: #c678dd;">plot</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"raw/{dataset}.csv"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"plots/{dataset}.pdf"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"somecommand {input} {output}"</span>
</pre>
            </div>
          </section>
          <section id="slide-orgf25898a">
            <h4 id="orgf25898a">Snakemake R</h4>
            <p>Requires <code>rpy2</code></p>
            <div class="org-src-container">
              <pre
                class="src src-snakemake"
              ><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">mapRNAseq</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #dcaeea;">file1</span> = <span style="color: #98be65;">"path/to/data/file1.bam"</span>,
        <span style="color: #dcaeea;">file2</span> = <span style="color: #98be65;">"path/to/data/file2.bam"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"path/to/qlf_table.csv"</span>
    <span style="color: #ECBE7B;">run</span>:
        R(<span style="color: #98be65;">"""</span>
<span style="color: #98be65;">        bamfiles &lt;- cbind({input.file1},{input.file2})</span>
<span style="color: #98be65;">        {output} &lt;- featureCounts(bamfiles, annot.inbuilt="hg19", strandspecific=2)</span>
<span style="color: #98be65;">        ...</span>
<span style="color: #98be65;">        ...</span>
<span style="color: #98be65;">        write.csv(qlf_table, file={output}, quote = FALSE)</span>
<span style="color: #98be65;">        """</span>)
</pre>
            </div>
          </section>
          <section id="slide-org3061251">
            <h4 id="org3061251">Snakemake R</h4>
            <div class="org-src-container">
              <pre
                class="src src-snakemake"
              ><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">GM19_genes_edgeR</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"results/2019-02-28/GM19_gene_counts.rds"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"results/2019-03-25/GM19_rawdata_table.csv"</span>,
        <span style="color: #98be65;">"results/2019-03-25/GM19_genelen.csv"</span>,
        <span style="color: #98be65;">"results/2019-03-25/GM19_normalized.txt"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"../../envs/edgeR.yaml"</span>
    <span style="color: #ECBE7B;">threads</span>: <span style="color: #da8548; font-weight: bold;">4</span>
    <span style="color: #ECBE7B;">script</span>:
        <span style="color: #98be65;">"../../scripts/dge.R"</span>
</pre>
            </div>
          </section>
          <section id="slide-org07c93a2">
            <h4 id="org07c93a2">Test data</h4>
            <div class="org-src-container">
              <pre class="src src-shell">
snakemake --use-conda --directory .test
</pre
              >
            </div>
          </section>
          <section id="slide-orgb6e1253">
            <h3 id="orgb6e1253">Organizing Bioinformatics Projects</h3>
            <div class="outline-text-3" id="text-orgb6e1253"></div>
          </section>
          <section
            id="slide-org458449f"
            data-background="img/org.png"
            data-background-size="1200px"
            data-background-transition="slide"
          >
            <h4 id="org458449f"></h4>
          </section>
        </section>
        <section>
          <section id="slide-org15791fc">
            <h2 id="org15791fc">HPC Cluster</h2>
            <div class="outline-text-2" id="text-org15791fc"></div>
          </section>
          <section id="slide-org579dd6b">
            <h3 id="org579dd6b">Intro to Ganymede</h3>
            <ul>
              <li><code>ganymedeadmins@utdallas.edu</code></li>
              <li>centos 7 with OpenHPC</li>
              <li>
                <a href="http://docs.oithpc.utdallas.edu"
                  >http://docs.oithpc.utdallas.edu</a
                >
              </li>
              <li>
                Access to <code>normal</code>, <code>debug</code>,
                <code>genomics</code>, and <code>GPU1</code> queue
              </li>
            </ul>
          </section>
          <section id="slide-org736eef8">
            <h3 id="org736eef8">Getting Access</h3>
            <ul>
              <li>
                Email <code>admin@ganymede</code> for access and to
                <b>genomics</b>
              </li>
              <li>Takes around a week</li>
              <li>
                You can log in with
                <code>ssh &lt;NETID&gt;@ganymede.utdallas.edu</code>
              </li>
            </ul>
          </section>
          <section id="slide-orgf004c09">
            <h3 id="orgf004c09">Some Basic Commands</h3>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><span style="color: #5B6268;"># </span><span style="color: #5B6268;">Info about slurm cluster</span>
sinfo
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">List enabled modules</span>
module list
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Available module</span>
module av
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Load anaconda</span>
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Must be done everytime you log in</span>
module load anaconda3
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Cancel all jobs</span>
scancel -u &lt;NETID&gt;
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Check the queue</span>
squeue -p genomics
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Submit a job</span>
sbatch --help
</pre>
            </div>
          </section>
          <section id="slide-org01f47d8">
            <h3 id="org01f47d8">Installing Conda</h3>
            <div class="org-src-container">
              <pre class="src src-shell">pip install conda
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">If that doesn't work</span>
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh

<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Make an environment</span>
conda create -n smk -c bioconda snakemake
conda install -c bioconda nextflow
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">OR</span>
conda create -n nf -c bioconda nextflow

<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Activate your env</span>
conda activate smk
</pre>
            </div>
          </section>
          <section id="slide-orgaede35a">
            <h3 id="orgaede35a">Running Things</h3>
            <ol>
              <li>
                Email <code>admin@ganymede</code> for access and to
                <b>genomics</b>
              </li>
              <li>Setup local cookiecutter</li>
              <li>Log in <code>ssh eam150030@ganymede.utdallas.edu</code></li>
              <li>Load Modules</li>
              <li>Rsync from local computer</li>
            </ol>
            <div class="org-src-container">
              <pre class="src src-shell">
rsync -avtr /media/enhancer/IMR90/data eam150030@ganymede.utdallas.edu:/home/eam150030/IMR90/data
</pre
              >
            </div>
            <ol>
              <li>Activate conda env <code>conda activate smk</code></li>
            </ol>
          </section>
          <section id="slide-org1ec2648">
            <h4 id="org1ec2648">Running Things</h4>
            <p>
              Basic script
            </p>
            <div class="org-src-container">
              <pre
                class="src src-bash"
              ><span style="color: #5B6268;">#</span><span style="color: #5B6268;">!/bin/</span><span style="color: #51afef;">bash</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --ntasks=1</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --time=00:10:00</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-user=eam150030@utdallas.edu</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-type=ALL</span>

bowtie2 file.fastq
...
rysnc output back
</pre>
            </div>
          </section>
          <section id="slide-org5052dff">
            <h3 id="org5052dff">Snakemake on Slurm</h3>
            <ol>
              <li>Have <code>snakemake</code> installed</li>
              <li>Make a cluster config file</li>
            </ol>
            <div class="org-src-container">
              <pre class="src src-json">{
    <span style="color: #51afef;">"__default__"</span> :
    {
        <span style="color: #51afef;">"account"</span> : <span style="color: #98be65;">"eam150030"</span>,
        <span style="color: #51afef;">"time"</span> : "<span style="color: #da8548; font-weight: bold;">00</span>:<span style="color: #da8548; font-weight: bold;">15</span>:<span style="color: #da8548; font-weight: bold;">00</span>",
        <span style="color: #51afef;">"n"</span> : <span style="color: #da8548; font-weight: bold;">1</span>,
        <span style="color: #51afef;">"partition"</span> : <span style="color: #98be65;">"genomics"</span>
    },
    <span style="color: #51afef;">"compute1"</span> :
    {
        <span style="color: #51afef;">"time"</span> : "<span style="color: #da8548; font-weight: bold;">00</span>:<span style="color: #da8548; font-weight: bold;">20</span>:<span style="color: #da8548; font-weight: bold;">00</span>"
    }
}
</pre>
            </div>
            <ol>
              <li>Use snakemake to submit jobs</li>
            </ol>
            <div class="org-src-container">
              <pre
                class="src src-bash"
              >snakemake -j <span style="color: #da8548; font-weight: bold;">999</span> --use-conda --cluster-config cluster.json <span style="color: #98be65;">\</span>
    --cluster <span style="color: #98be65;">"sbatch -A {cluster.account} -p {cluster.partition}\</span>
<span style="color: #98be65;">    -n {cluster.n}  -t {cluster.time}"</span>
</pre>
            </div>
          </section>
          <section id="slide-org08891fd">
            <h4 id="org08891fd">Slurm Script</h4>
            <div class="org-src-container">
              <pre
                class="src src-bash"
              ><span style="color: #5B6268;">#</span><span style="color: #5B6268;">!/bin/</span><span style="color: #51afef;">bash</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --ntasks=1</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --time=00:10:00</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-user=eam150030@utdallas.edu</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-type=ALL</span>

conda activate smk

snakemake -j <span style="color: #da8548; font-weight: bold;">999</span> --use-conda --cluster-config cluster.json <span style="color: #98be65;">\</span>
    --cluster <span style="color: #98be65;">"sbatch -A {cluster.account} -p {cluster.partition}\</span>
<span style="color: #98be65;">    -n {cluster.n}  -t {cluster.time}"</span>

rysnc -avtr results myLabComputer
</pre>
            </div>
          </section>
          <section id="slide-org9517be2">
            <h3 id="org9517be2">Nextflow</h3>
            <ol>
              <li>Have <code>nextflow</code> installed</li>
              <li>Running a toy example</li>
            </ol>
            <div class="org-src-container">
              <pre class="src src-shell">
nextflow run rnatoy -with-singularity
</pre
              >
            </div>
            <ul>
              <li>
                Ganymede doesn't have docker so you can't use
                <code>-with-docker</code>
              </li>
            </ul>
            <ol>
              <li>
                Executing using slurm (<a
                  href="https://www.nextflow.io/docs/latest/executor.html?highlight=slurm#slurm"
                  >docs</a
                >)
              </li>
            </ol>
            <div class="org-src-container">
              <pre
                class="src src-shell"
              ><span style="color: #5B6268;"># </span><span style="color: #5B6268;">nextflow.config</span>
process <span style="color: #51afef;">{</span>
  <span style="color: #dcaeea;">executor</span> = <span style="color: #98be65;">'slurm'</span>
  <span style="color: #dcaeea;">queue</span> = <span style="color: #98be65;">'genomics'</span>
<span style="color: #51afef;">}</span>
</pre>
            </div>
            <ol>
              <li>Run pipeline</li>
            </ol>
            <div class="org-src-container">
              <pre class="src src-shell">
nextflow run tutorial.nf
nextflow -c nextflow.config run rnatoy -with-singularity
</pre
              >
            </div>
          </section>
          <section id="slide-org587b392">
            <h3 id="org587b392">PyTorch</h3>
            <div class="org-src-container">
              <pre
                class="src src-python"
              ><span style="color: #51afef;">from</span> __future__ <span style="color: #51afef;">import</span> print_function
<span style="color: #51afef;">import</span> torch
torch.cuda.is_available()
<span style="color: #dcaeea;">x</span> = torch.rand(<span style="color: #da8548; font-weight: bold;">5</span>, <span style="color: #da8548; font-weight: bold;">3</span>)
<span style="color: #51afef;">print</span>(x)
</pre>
            </div>
          </section>
          <section id="slide-orgae3abf6">
            <h4 id="orgae3abf6">PyTorch Slurm Script</h4>
            <div class="org-src-container">
              <pre
                class="src src-bash"
              ><span style="color: #5B6268;">#</span><span style="color: #5B6268;">!/usr/bin/</span><span style="color: #51afef;">env</span><span style="color: #5B6268;"> bash</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -J pytorchtest</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -o pytorchtest-%A.out</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -e pytorchtest-%A.err</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -p GPU1</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --gres=gpu:1</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -c 1</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH -t 00:01:00</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-user=eam150030@utdallas.edu</span>
<span style="color: #5B6268;">#</span><span style="color: #5B6268;">SBATCH --mail-type=ALL</span>

module purge
module load singularity
module load CUDA
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Assuming that the container has been copied to the user's /scratch directory</span>
singularity exec docker://pytorch/pytorch python <span style="color: #98be65;">\</span>
    /home/eam150030/pytorch-demo/pytorch_example.py
</pre>
            </div>
          </section>
        </section>
        <section>
          <section id="slide-orga718b69">
            <h2 id="orga718b69">
              eRNA Prediction Pipeline from GRO-Seq INFB1 Induction Timecourse
            </h2>
            <div class="outline-text-2" id="text-orga718b69"></div>
          </section>
          <section
            id="slide-orgb066d44"
            data-background="img/globaltrans.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h3 id="orgb066d44"></h3>
          </section>
          <section id="slide-org9c3b99c">
            <h3 id="org9c3b99c">Overview</h3>
            <ul>
              <li>Reproducing GM18</li>
              <li>Predicted IMR90 eRNAs</li>
              <li>Compared IMR90 Predicted Enhancers to GM</li>
              <li>Used Homer scripts to find DE of eRNAs and Genes</li>
              <li>Gene Centric vs. Enhancer Centric</li>
            </ul>
          </section>
          <section id="slide-org4b0379c">
            <h3 id="org4b0379c">Reproducing GM18</h3>
            <ul>
              <li>hg18 vs hg19</li>
              <li>Overpredicting eRNA transcripts</li>
              <li>
                Past Issue
                <ul>
                  <li>What I thought Peng sent me</li>
                  <li>hg18 -&gt; eRNAs -&gt; Me</li>
                  <li>What actually happened</li>
                  <li>hg18 -&gt; eRNAs -&gt; LiftOver -&gt; hg19 -&gt; Me</li>
                </ul>
              </li>
              <li>Main issue is homer uniqmap</li>
            </ul>
          </section>
          <section
            id="slide-org1b1454e"
            data-background="img/Figure_1.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org1b1454e"></h4>
          </section>
          <section
            id="slide-org290fdbd"
            data-background="img/Figure_2.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org290fdbd"></h4>
          </section>
          <section
            id="slide-orgec4e143"
            data-background="img/dag.png"
            data-background-size="400px"
            data-background-transition="slide"
          >
            <h4 id="orgec4e143"></h4>
          </section>
          <section id="slide-org3c32cf5">
            <h4 id="org3c32cf5">Actual Pipelines</h4>
            <div class="outline-text-4" id="text-org3c32cf5"></div>
            <ul class="org-ul">
              <li>
                <a id="org7e59335"></a><a href="files/GM18dag.pdf">GM18</a
                ><br />
              </li>
              <li>
                <a id="org28121c7"></a><a href="files/IMR19dag.pdf">IMR19</a
                ><br />
              </li>
              <li>
                <a id="org7547afe"></a><a href="files/all.pdf">All</a><br />
              </li>
            </ul>
          </section>
          <section id="slide-org5774f7e">
            <h3 id="org5774f7e">Predicted IMR90 eRNAs</h3>
            <p>
              Changes from GM18
            </p>
            <ul>
              <li>hg19</li>
              <li>No liftover</li>
            </ul>
          </section>
          <section id="slide-orge9319e4">
            <h3 id="orge9319e4">Compared IMR90 Predicted Enhancers to GM</h3>
            <div class="outline-text-3" id="text-orge9319e4"></div>
          </section>
          <section
            id="slide-org448c882"
            data-background="img/Venn.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org448c882"></h4>
          </section>
          <section id="slide-org23a4635">
            <h3 id="org23a4635">
              Used Homer scripts to find DE of eRNAs and Genes
            </h3>
          </section>
          <section id="slide-orged6a42f">
            <h3 id="orged6a42f">Gene Centric vs. Enhancer Centric</h3>
            <ul>
              <li>
                Peng's approach
                <ul>
                  <li>Took enhancers that were expressed deferentially</li>
                  <li>Linked them to Genes within 200Kb</li>
                </ul>
              </li>
              <li>
                New approach
                <ul>
                  <li>Take genes that are deferentially expressed</li>
                  <li>Link the Enhancers to those genes</li>
                </ul>
              </li>
            </ul>
          </section>
          <section
            id="slide-org3e286ce"
            data-background="viz/pipeline.png"
            data-background-size="800px"
            data-background-transition="slide"
          >
            <h4 id="org3e286ce"></h4>
          </section>
          <section id="slide-org23a8b37">
            <h4 id="org23a8b37">featureCounts</h4>
            <div class="outline-text-4" id="text-org23a8b37"></div>
            <ul class="org-ul">
              <li><a id="org54ad195"></a>Tried CLI version<br /></li>
              <li>
                <a id="org5836012"></a>Couldn't get examples to work with
                edgeR<br />
              </li>
              <li><a id="orge737391"></a>Switched to R package<br /></li>
            </ul>
          </section>
          <section id="slide-org8ca5975">
            <h4 id="org8ca5975">edgeR</h4>
            <div class="outline-text-4" id="text-org8ca5975"></div>
            <ul class="org-ul">
              <li><a id="org3f42cd2"></a>Error with exactTest<br /></li>
              <li>
                <a id="orgc1e8043"></a>Defaulted to writing raw DGEList to
                file<br />
              </li>
              <li><a id="org38382aa"></a>Used in Pivot<br /></li>
            </ul>
          </section>
          <section id="slide-org9b9e982">
            <h4 id="org9b9e982">Results</h4>
          </section>
          <section
            id="slide-org4c0943c"
            data-background="img/GM_normalized.png"
            data-background-size="1200px"
            data-background-transition="slide"
          >
            <h4 id="org4c0943c"></h4>
          </section>
          <section
            id="slide-org54b219c"
            data-background="img/IMR_normalized.png"
            data-background-size="1200px"
            data-background-transition="slide"
          >
            <h4 id="org54b219c"></h4>
          </section>
          <section id="slide-org6d72b3e">
            <h4 id="org6d72b3e">Takeaways</h4>
            <ul>
              <li>Divide by median read values</li>
              <li>Need to go from -2 to 2 to show kinetics</li>
            </ul>
          </section>
          <section id="slide-org2cd355f">
            <h3 id="org2cd355f">Future</h3>
            <ul>
              <li>Kinetics of the eRNAs</li>
              <li>Compare IMR and GM Genes</li>
              <li>Get List of Top Genes</li>
              <li>Link eRNAs and Top genes</li>
            </ul>
          </section>
        </section>
      </div>
    </div>
    <script src="http://cdn.jsdelivr.net/reveal.js/3.0.0/lib/js/head.min.js"></script>
    <script src="http://cdn.jsdelivr.net/reveal.js/3.0.0/js/reveal.js"></script>

    <script>
      // Full list of configuration options available here:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        controls: true,
        progress: true,
        history: false,
        center: true,
        slideNumber: "c",
        rollingLinks: false,
        keyboard: true,
        overview: true,

        theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
        transition: Reveal.getQueryHash().transition || "default", // default/cube/page/concave/zoom/linear/fade/none
        transitionSpeed: "default",
        multiplex: {
          secret: "", // null if client
          id: "", // id, obtained from socket.io server
          url: "" // Location of socket.io server
        },

        // Optional libraries used to extend on reveal.js
        dependencies: [
          {
            src: "http://cdn.jsdelivr.net/reveal.js/3.0.0/lib/js/classList.js",
            condition: function() {
              return !document.body.classList;
            }
          },
          {
            src:
              "http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/markdown/marked.js",
            condition: function() {
              return !!document.querySelector("[data-markdown]");
            }
          },
          {
            src:
              "http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/markdown/markdown.js",
            condition: function() {
              return !!document.querySelector("[data-markdown]");
            }
          },
          {
            src:
              "http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/zoom-js/zoom.js",
            async: true,
            condition: function() {
              return !!document.body.classList;
            }
          },
          {
            src:
              "http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/notes/notes.js",
            async: true,
            condition: function() {
              return !!document.body.classList;
            }
          }
        ]
      });
    </script>
  </body>
</html>
