<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>Gro-Seq</title>
<meta name="author" content="(Edmund Miller)"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="http://cdn.jsdelivr.net/reveal.js/3.0.0/css/reveal.css"/>

<link rel="stylesheet" href="http://cdn.jsdelivr.net/reveal.js/3.0.0/css/theme/white.css" id="theme"/>


<!-- If the query includes 'print-pdf', include the PDF print sheet -->
<script>
    if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'http://cdn.jsdelivr.net/reveal.js/3.0.0/css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
</script>
</head>
<body>
<div class="reveal">
<div class="slides">
<section>
<section id="slide-org97a7b6b">
<h2 id="org97a7b6b">GRO-Seq Analysis</h2>
<div class="outline-text-2" id="text-org97a7b6b">
</div>
</section>
</section>
<section>
<section id="slide-orgafef1f8">
<h3 id="orgafef1f8">IMR90</h3>
<div class="outline-text-3" id="text-orgafef1f8">
</div>
</section>
<section id="slide-orgd10dd2a">
<h4 id="orgd10dd2a">Recap</h4>
<ul>
<li>Predicted IMR90 eRNAs</li>
<li>Compared IMR90 Predicted Enhancers to GM</li>
<li>Ran GREAT Analysis on them</li>
<li>Began working to find coverage of eRNAs in Samples</li>

</ul>
</section>
</section>
<section>
<section id="slide-orge621089">
<h3 id="orge621089">What I've been working on</h3>
<div class="outline-text-3" id="text-orge621089">
</div>
</section>
<section id="slide-org314edd6">
<h4 id="org314edd6">Different Liftover points</h4>
</section>
<section id="slide-org8912ec3" data-background="img/dag.png" data-background-size="400px" data-background-transition="slide">
<h4 id="org8912ec3">.</h4>
</section>
<section id="slide-org7d5048d">
<h4 id="org7d5048d">Uniqmap</h4>
<ul>
<li>Since some transcripts cover very large regions, there are many places where genomic repeats interrupt the GRO-Seq signal of continous transcripts.  To help deal with this problem, HOMER can take advantage of mappability information to help estimate transcript levels where uniquely mapping sequencing reads is not possible.</li>
<li>hg19 is provided by homer</li>
<li>hg18 you have to compile yourself</li>

</ul>
</section>
<section id="slide-orgdd11351">
<h4 id="orgdd11351">Differential Expression</h4>
<ul>
<li><code>getDifferentialPeaksReplicates.pl</code> - perform peak finding/Differential peak detection taking into account replicates</li>
<li>Second attempt
<ul>
<li><code>annotatePeaks.pl</code> - annotation of genomic positions, organization of motif and sequencing data, histograms, heatmaps,</li>
<li><code>getDifferentialPeaks</code> - find differentially bound regions from ChIP/ATAC-seq experiments</li>

</ul></li>
<li>Third attempt
<ul>
<li><code>analyzeRepeats.pl</code> - quantification of RNA levels across repeats</li>
<li><code>getDiffExpression.pl</code> - Calculate differential enrichment of RNA-seq/ChIP-seq/ATAC-seq data</li>

</ul></li>

</ul>
</section>
<section id="slide-org23354d9">
<h4 id="org23354d9">Issues</h4>
<ul>
<li>Couldn't get Deseq2 to work</li>
<li>Homer recommends:
<ul>
<li><code>In general, if you don't have replicates, I would recommend running the program with the paramters "-edgeR -simpleNorm -dispersion 0.05".</code></li>

</ul></li>

</ul>
</section>
</section>
<section>
<section id="slide-orgc694015">
<h3 id="orgc694015">Results</h3>
<div class="outline-text-3" id="text-orgc694015">
</div>
</section>
<section id="slide-org52ee457" data-background="img/Venn.png" data-background-size="800px" data-background-transition="slide">
<h4 id="org52ee457">.</h4>
</section>
<section id="slide-orgc4eaebc" data-background="img/Meta.png" data-background-size="800px" data-background-transition="slide">
<h4 id="orgc4eaebc">.</h4>
</section>
<section id="slide-org765c67a" data-background="img/GM_eRNAs.png" data-background-size="800px" data-background-transition="slide">
<h4 id="org765c67a">.</h4>
</section>
<section id="slide-orge017f7c" data-background="img/IMR_eRNAs.png" data-background-size="800px" data-background-transition="slide">
<h4 id="orge017f7c">.</h4>
</section>
<section id="slide-orgb101fbb" data-background="img/GM_eRNAs_overlap.png" data-background-size="800px" data-background-transition="slide">
<h4 id="orgb101fbb">.</h4>
</section>
<section id="slide-org371b3c3" data-background="img/IMR_eRNAs_overlap.png" data-background-size="800px" data-background-transition="slide">
<h4 id="org371b3c3">.</h4>
</section>
</section>
<section>
<section id="slide-orge13cb2a">
<h3 id="orge13cb2a">Future</h3>
<div class="outline-text-3" id="text-orge13cb2a">
</div>
</section>
<section id="slide-org6b62465">
<h4 id="org6b62465">Gene Centric vs. Enhancer Centric</h4>
<ul>
<li>Peng's approach
<ul>
<li>Took enhancers that were expressed deferentially</li>
<li>Linked them to Genes within 200Kb</li>

</ul></li>
<li>New approach
<ul>
<li>Take genes that are deferentially expressed</li>
<li>Link the Enhancers to those genes</li>

</ul></li>

</ul>
</section>
<section id="slide-orgf0ce52e">
<h4 id="orgf0ce52e">So Far</h4>
<ul>
<li>Getting about 8% of 65572 genes of IMR90 up or down regulated per hour</li>
<li>GM is less than 1% for 65569 genes up or down regulated per hour</li>
<li>Need to understand <a href="https://davetang.org/muse/2011/07/27/edgers-common-dispersion/">Common dispersion</a></li>

</ul>
</section>
</section>
</div>
</div>
<script src="http://cdn.jsdelivr.net/reveal.js/3.0.0/lib/js/head.min.js"></script>
<script src="http://cdn.jsdelivr.net/reveal.js/3.0.0/js/reveal.js"></script>

<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({

controls: true,
progress: true,
history: false,
center: true,
slideNumber: 'c',
rollingLinks: false,
keyboard: true,
overview: true,

theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none
transitionSpeed: 'default',
multiplex: {
    secret: '', // null if client
    id: '', // id, obtained from socket.io server
    url: '' // Location of socket.io server
},

// Optional libraries used to extend on reveal.js
dependencies: [
 { src: 'http://cdn.jsdelivr.net/reveal.js/3.0.0/lib/js/classList.js', condition: function() { return !document.body.classList; } },
 { src: 'http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'http://cdn.jsdelivr.net/reveal.js/3.0.0/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }]
});
</script>
</body>
</html>
