#+REVEAL_THEME: white
#+TITLE: Xonsh, Nix, & Limma
#+AUTHOR: Edmund Miller
#+OPTIONS: reveal_title_slide:nil
#+OPTIONS: num:nil
#+OPTIONS: toc:nil
#+REVEAL_ROOT: https://cdn.jsdelivr.net/npm/reveal.js
#+REVEAL_HLEVEL: 1

* Xonsh
** SciPy 2019
*** Links
**** [[http://conference.scipy.org/][Conference List]]
**** [[https://xonsh.github.io/scipy-2019-tutorial/#1][Xonsh Tutorial]]
**** [[https://youtu.be/ujo9sbqHFqw?list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC][Xonsh Talk]]
**** [[https://www.youtube.com/watch?v=ZB7BZMhfPgk&list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC][Talk Playlist]]
* Nix
** Code as Infastructure
- [[https://nixos.org/~eelco/pubs/nixos-jfp-final.pdf][NixOS: A Purely Functional Linux Distribution]]
** A Couple of Different Things
- Nix Programming Language
- NIx Package Manager
- NixOS
- [[https://www.youtube.com/watch?v=USDbjmxEZ_I&list=PLYx7XA2nY5GcDQblpQ_M1V3PQPoLWiDAC&index=16][SciPy Talk]]
** 
:PROPERTIES:
:reveal_background: img/HowNix.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
** Simple Derivation
#+BEGIN_SRC sh
declare -xp
export PATH="$coreutils/bin"
echo "hello"
echo $blah
dest=$out/bin
mkdir -p $dest
cp $src $dest/my-hello
#+END_SRC
** Simple Derivation
#+BEGIN_SRC nix
with (import <nixpkgs> {});
derivation {
  name = "my_hello";
  builder = "${bash}/bin/bash";
  args = [ ./build.sh ];
  system = builtins.currentSystem;
  coreutils = coreutils;
  src = ./my-hello;
  blah = ./blah;
}
#+END_SRC
** My Config
** Python and Nix
*** Imperative Installation
#+BEGIN_SRC shell
$ nix-env -i hello
$ nix-env --list-generations
$ nix-env --rollback
# View environment dependencies
$ nix-env -q
#+END_SRC
*** nix-shell
#+BEGIN_SRC shell
$ nix-shell --pure -p python27 -p python27Packages.numpy
#+END_SRC
#+BEGIN_SRC shell
$ python -c "import numpy"
$ echo $PYTHONPATH
#+END_SRC
*** Repeatable Environment
#+BEGIN_SRC nix
# shell.nix
let
  nixpkgs = import <nixpkgs> {};
  pypkgs = nixpks.python36Packages;
in
  (pypkgs.python.withPackages (x: [ x.numpy x.scipy])).env
#+END_SRC
- Can also prevent version drift
*** buildPythonPackage
#+BEGIN_SRC nix
{ nixpkgs, pypkgs }:
pypkgs.buildPythonPackage rec {
  pname = "scipy";
  version = "0.19.1";
  src = pypkgs.fetchPypi {
    inherit pname version;
    sha256 = "1rl411bvla6q7qfdb47fpdnyjhfgzl6smpha33n9ar1klykjr6m1";
  };
  buildInputs = [
    pypkgs.numpy
    nixpkgs.pkgs.gcc
    nixpkgs.pkgs.gfortran
  ];
  ignoreCollisions=true;
  catchConflicts=false;
  doCheck = false;
}
#+END_SRC
- Fetch from PyPi
- Specify build time, run time, tests, dependancies
- Runs ~setup.py install~ or ~setup.py develop~
*** PyPi2Nix
#+BEGIN_SRC shell
$ pypi2nix -V "3.6" -r requirements.txt
$ nix-shell --pure requirements.nix
#+END_SRC
- Doesn't matter if packages aren't in nix packages
* eRNA Prediction Pipeline from GRO-Seq INFB1 Induction Timecourse
** Recap
*** 
:PROPERTIES:
:reveal_background: img/globaltrans.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** Overview
- Reproducing GM18
- Predicted IMR90 eRNAs
- Compared IMR90 Predicted Enhancers to GM
- Used Homer scripts to find DE of eRNAs and Genes
- Gene Centric vs. Enhancer Centric
*** Reproducing GM18
- hg18 vs hg19
- Overpredicting eRNA transcripts
- Past Issue
  - What I thought Peng sent me
  - hg18 -> eRNAs -> Me
  - What actually happened
  - hg18 -> eRNAs -> LiftOver -> hg19 -> Me
- Main issue is homer uniqmap
*** 
:PROPERTIES:
:reveal_background: img/Figure_1.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/Figure_2.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/dag.png
:reveal_background_size: 400px
:reveal_background_trans: slide
:END:
*** Actual Pipelines
***** [[file:files/GM18dag.pdf][GM18]]
***** [[file:files/IMR19dag.pdf][IMR19]]
***** [[file:files/all.pdf][All]]
*** Predicted IMR90 eRNAs
Changes from GM18
- hg19
- No liftover
*** Compared IMR90 Predicted Enhancers to GM
*** 
:PROPERTIES:
:reveal_background: img/Venn.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:

*** Used Homer scripts to find DE of eRNAs and Genes
*** Gene Centric vs. Enhancer Centric
- Peng's approach
  - Took enhancers that were expressed deferentially
  - Linked them to Genes within 200Kb
- New approach
  - Take genes that are deferentially expressed
  - Link the Enhancers to those genes

*** 
:PROPERTIES:
:reveal_background: viz/pipeline.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** featureCounts
***** Tried CLI version
***** Couldn't get examples to work with edgeR
***** Switched to R package
***** Update: switched back
** New Things
*** Shayne Easterwood
- NOISeq Analysis
- SAF Format for eRNA formats
- A chance to clean up
*** Linear Model / Time Series Analysis
*** Limma
- [[https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937821/][RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR]]
- [[http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MDS-Plot.html][glimma Plots]]
*** 
:PROPERTIES:
:reveal_background: img/limma/GM19_fig1_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/IMR_fig1_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/GM19_fig2_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/IMR_fig2_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/GM19_fig3_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/limma/IMR_fig3_limma.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** Future
- Get List of Top Genes
- Link eRNAs and Top genes
- Compare IMR and GM Genes
- Kinetics of the eRNAs
