#+TITLE: Gro-Seq
#+AUTHOR: Edmund Miller
#+OPTIONS: reveal_title_slide:nil
#+OPTIONS: num:nil
#+OPTIONS: toc:nil
#+OPTIONS: DATE:false
#+REVEAL_THEME: white
#+REVEAL_ROOT: https://cdn.jsdelivr.net/npm/reveal.js
#+REVEAL_HLEVEL: 2
* GRO-Seq Analysis
** IMR90
*** Recap
- Predicted IMR90 eRNAs
- Compared IMR90 Predicted Enhancers to GM
- Ran GREAT Analysis on them
- Began working to find coverage of eRNAs in Samples
** What I've been working on
*** Different Liftover points
*** 
:PROPERTIES:
:reveal_background: img/dag.png
:reveal_background_size: 400px
:reveal_background_trans: slide
:END:
*** Uniqmap
- Since some transcripts cover very large regions, there are many places where genomic repeats interrupt the GRO-Seq signal of continous transcripts.  To help deal with this problem, HOMER can take advantage of mappability information to help estimate transcript levels where uniquely mapping sequencing reads is not possible.
- hg19 is provided by homer
- hg18 you have to compile yourself
*** Differential Expression
- ~getDifferentialPeaksReplicates.pl~ - perform peak finding/Differential peak detection taking into account replicates
- Second attempt
  - ~annotatePeaks.pl~ - annotation of genomic positions, organization of motif and sequencing data, histograms, heatmaps,
  - ~getDifferentialPeaks~ - find differentially bound regions from ChIP/ATAC-seq experiments
- Third attempt
  - ~analyzeRepeats.pl~ - quantification of RNA levels across repeats
  - ~getDiffExpression.pl~ - Calculate differential enrichment of RNA-seq/ChIP-seq/ATAC-seq data
*** Issues
- Couldn't get Deseq2 to work
- Homer recommends:
  - ~In general, if you don't have replicates, I would recommend running the program with the paramters "-edgeR -simpleNorm -dispersion 0.05".~
** Results
*** 
:PROPERTIES:
:reveal_background: img/Venn.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/Meta.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/GM_eRNAs.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/IMR_eRNAs.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
*** 
:PROPERTIES:
:reveal_background: img/GM_eRNAs_overlap.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
***
:PROPERTIES:
:reveal_background: img/IMR_eRNAs_overlap.png
:reveal_background_size: 800px
:reveal_background_trans: slide
:END:
** Future
*** Gene Centric vs. Enhancer Centric
- Peng's approach
  - Took enhancers that were expressed deferentially
  - Linked them to Genes within 200Kb
- New approach
  - Take genes that are deferentially expressed
  - Link the Enhancers to those genes
*** So Far
- Getting about 8% of 65572 genes of IMR90 up or down regulated per hour
- GM is less than 1% for 65569 genes up or down regulated per hour
- Need to understand [[https://davetang.org/muse/2011/07/27/edgers-common-dispersion/][Common dispersion]]
